archive/tar_test.go

```diff
@@ -35,24 +36,40 @@ func testTar(t *testing.T, when spec.G, it spec.S) {
 		h.AssertNil(t, os.RemoveAll(tmpDir))
 	})
 
-	when("#ReadTarArchive", func() {
+	when("#UntarLayer", func() {
 		var pathModes = []archive.PathMode{
 			{"root", os.ModeDir + 0755},
 			{"root/readonly", os.ModeDir + 0500},
 			{"root/standarddir", os.ModeDir + 0755},
 			{"root/standarddir/somefile", 0644},
 			{"root/readonly/readonlysub/somefile", 0444},
 			{"root/readonly/readonlysub", os.ModeDir + 0500},
+			{"Files", os.ModeDir + 0755},
+			{"Files/standardfile_Files", 0644},
+			{"Hives", os.ModeDir + 0755},
+			{"Hives/standardfile_Hives", 0644},
+		}
+
+		// Golang for Windows only implements owner permissions
+		if runtime.GOOS == "windows" {
+			pathModes = []archive.PathMode{
+				{"root", os.ModeDir + 0777},
+				{"root/readonly", os.ModeDir + 0555},
+				{"root/standarddir", os.ModeDir + 0777},
+				{"root/standarddir/somefile", 0666},
+				{"root/readonly/readonlysub/somefile", 0444},
+				{"root/readonly/readonlysub", os.ModeDir + 0555},
+				{"standardfile_Files", 0666},
+				{"standardfile_Hives", 0666},
```

Comment:
After talking with @micahyoung  in buildpacks slack it seems like this isn't a realistic test value and documents an incorrect handling actual `Hives/*` entries

> Micah:
There are five “Hive-deltas” in the Hives/* part of the layer and each one gets merged into one of 5 registry files that appear on-disk inside the container
>
>A base image has most of the entries for one of those sections, but any additional layers will add more entries
>
>The names are:
Hives/DefaultUser_Delta
Hives/Sam_Delta
Hives/Security_Delta
Hives/Software_Delta
Hives/System_Delta

Seems like we should either merge `Hive` deltas if we find them, or (if we want to punt for now) just ignore `Hives/*` files?

Reply:
I think that merging Hive deltas can be saved for a later PR. I'll adjust the code to skip for now, and we can submit a PR when we want to address it.

Reply:
OK update is pushed @ekcasey